js/views/userPage/Store.js

```diff
@@ -32,15 +35,85 @@ export default class extends BaseVw {
       freeShipping: false,
     };
 
+    this.listingsViewType = app.localSettings.get('listingsGridViewType');
+
     this.listenTo(this.collection, 'request', this.onRequest);
     this.listenTo(this.collection, 'update', this.onUpdateCollection);
 
+    if (this.model.id === app.profile.id) {
+      this.listenTo(listingEvents, 'saved', (md, opts) => {
+        // For now, we only know if the listing model has
+        // changed in some way since the last save. We don't
+        // know what specifically changed. So, this message
+        // will show if some listing attribute changed, even
+        // though it may not be one represented in the store.
+        if (opts.hasChanged()) {
+          this.showDataChangedMessage();
+        }
+      });
+
+      this.listenTo(listingEvents, 'destroy', () => (this.showDataChangedMessage()));
+    }
+
+    this.listenTo(app.settings, 'change:country', () => (this.showShippingChangedMessage()));
+
+    this.listenTo(app.settings.get('shippingAddresses'), 'update',
+      (cl, opts) => {
+        if (opts.changes.added.length ||
+          opts.changes.removed.length) {
+          this.showShippingChangedMessage();
+        }
+      });
+
+    // this block should be last
     if (options.initialFetch) {
       this.fetch = options.initialFetch;
       this.onRequest(this.collection, this.fetch);
     }
   }
 
+  showDataChangedMessage() {
+    if (this.dataChangePopIn && !this.dataChangePopIn.isRemoved()) {
+      this.dataChangePopIn.$el.velocity('callout.shake', { duration: 500 });
+    } else {
+      this.dataChangePopIn = this.createChild(PopInMessage, {
+        messageText: 'Listing data has changed (translate me). ' +
+          '<a class="js-refresh">refresh</a>',
+      });
+
+      this.listenTo(this.dataChangePopIn, 'clickRefresh', () => (this.collection.fetch()));
+
+      this.listenTo(this.dataChangePopIn, 'clickDismiss', () => {
+        this.dataChangePopIn.remove();
+        this.dataChangePopIn = null;
+      });
+
+      this.$popInMessages.append(this.dataChangePopIn.render().el);
+    }
+  }
+
+  showShippingChangedMessage() {
+    if (this.shippingChangePopIn ||
+      (this.shippingChangePopIn && this.shippingChangePopIn.isRemoved())) {
```

Comment:
Similar comment here. When would this.shippingChangePopIn be false, but (this.shippingChangePopIn && this.shippingChangePopIn.isRemoved()) be true?